src/wp-includes/class-wp-scripts.php

```diff
@@ -284,29 +302,27 @@ public function do_item( $handle, $group = false ) {
 			$ver = $ver ? $ver . '&amp;' . $this->args[ $handle ] : $this->args[ $handle ];
 		}
 
-		$src         = $obj->src;
-		$cond_before = '';
-		$cond_after  = '';
-		$conditional = isset( $obj->extra['conditional'] ) ? $obj->extra['conditional'] : '';
+		$src               = $obj->src;
+		$strategy          = $this->get_eligible_loading_strategy( $handle );
+		$intended_strategy = $this->get_data( $handle, 'strategy' );
+		$cond_before       = '';
+		$cond_after        = '';
+		$conditional       = isset( $obj->extra['conditional'] ) ? $obj->extra['conditional'] : '';
+
+		if ( ! $this->is_delayed_strategy( (string) $intended_strategy ) ) {
```

Comment:
Should `$intended_strategy` be cast to `string` _here_, or when it's initialized?

Reply:
Good question. It's only being cast here because `is_delayed_strategy` expects a string and the `get_data` method can return a `false` boolean value if no strategy is found. However, from this point on, every part of the code has to assume that a `false` may have been converted to an empty string anyway, so I think it would be fine to cast it sooner, and might be a bit cleaner.

Suggestion applied in [dc1b203](https://github.com/WordPress/wordpress-develop/pull/4391/commits/dc1b20329f83ab55c169b70f5f1e9e9076240828).